broker-daemon/block-order-worker/index.js

```diff
@@ -91,7 +114,22 @@ class BlockOrderWorker extends EventEmitter {
 
     this.logger.info(`Created and stored block order`, { blockOrderId: blockOrder.id })
 
-    this.emit('BlockOrder:create', blockOrder)
+    // Register events on a block order worker to handle processing of concurrent orders
+    const createEventId = generateIdForEvent(WORKER_EVENTS.CREATE, id)
+    const rejectedEventId = generateIdForEvent(WORKER_EVENTS.REJECTED, id)
+
+    // Move BlockOrder to a rejected state if any process in working a block order, fails
+    this.once(rejectedEventId, async (blockOrderId, err) => this.failBlockOrder(blockOrderId, err))
+
+    // Start working the block order in another process to prevent blocking the creation
+    // of 'other' block orders
+    this.once(createEventId, async (blockOrder) => {
+      await this.workBlockOrder(blockOrder).catch(e => this.emit(rejectedEventId, blockOrder.id, e))
+    })
+
+    // TODO: Use an ID instead of the entire BlockOrder so that we can have multiple
```

Comment:
I think its a good practice to pass IDs around instead of objects because the object can be mutated before the event occurs (which could lead to issues). ill update the comment